### PR TITLE
Support Expr.toString and add Json serialization for debugging

### PR DESCRIPTION
**Pull request by jeffmay**
- Add DisplayExpr and make it the toString of Expr
- Add InterpretExprAsJson and InterpretExprResultAsJson for debugging

Depends on:
- [x] https://github.com/jeffmay/vapors/pull/25

### CODE REVIEW

**Comment by kionka-rally** [comment]
Honestly, this is pretty hard to review with all the codecov warnings. It would be nice if there were an easy to disable them. That said, it all seems reasonable to me.

File: core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/DisplayExprSpec.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    }
+
+    "serializing with newlines and 2 spaces" should {
+
+      "print a map and select from list of facts" in {
+        val q = withFactsOfType(FactTypes.Age).where { facts =>
+          facts.toList.map(_.get(_.select(_.value)))
+        }
+        pendingUntilFixed {
```
Comment (kionka-rally) [inline core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/DisplayExprSpec.scala:31]: I'm guessing this PR is a work in progress?

Comment (kionka-rally) [inline core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/DisplayExprSpec.scala:32]: I'm guessing this PR is a work in progress?

**Comment by jeffmay** [comment]
Closing this in favor of the new algebra which will replace the code this is based on. The PRs start with https://github.com/jeffmay/vapors/pull/69

**Pull request closed without merging**